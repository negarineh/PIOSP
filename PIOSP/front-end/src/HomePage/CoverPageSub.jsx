import React, { Component } from 'react';
import PropTypes from 'prop-types';
import '../components/stylesheets/MainPage.css';

import { Segment, Image, Icon, Card, Divider, Grid, Label, Menu } from 'semantic-ui-react';
import '../../node_modules/semantic-ui-css/semantic.min.css';

import {SurveyResultsPage} from '../SurveyResultsPage';
import {NumberOfCorrectAnswersCategory} from '../SurveyResultsPage';
import {ReportAnswerCategoryPage} from '../ReportAnswerCategoryPage';

// customized cover page for sub admins
class CoverPagesSub extends Component {

  constructor(props) {
    super(props);
    this.state = {
      activeItem: 'home',
      visible: false,
    };
    this.handleItemClick = this.handleItemClick.bind(this);
  }

  toggleVisibility = () => this.setState({ visible: !this.state.visible })

  handleItemClick = (e, { name }) => {
    this.setState({ activeItem: name });
  }

  handleChange = (e, { name, value }) => 
  {
    this.setState({ [name]: value });
  }


  render() {
    const {activeItem} = this.state;
    return (
      <div >
      
          <Segment style={{background: 'whitesmoke', height:'638.5px'}}>
          {activeItem === 'home'?
              <Grid centered container columns={3}>
              <Grid.Column>
              <Card>
              <Label as='a' corner='left' style={{color:'rgb(92,179,96)'}}></Label>
                <Card.Content>
                  <Card.Header style={{paddingLeft:'20px'}}>
                    Graphs
                  </Card.Header>
                  <Divider horizontal/>
                    <Image rounded src="/img/images.jpeg" size='small' />
                  <Divider horizontal/>
                </Card.Content>
                <Card.Content >
                  <Menu secondary size='tiny'>
                    <Menu.Item name='visulize' onClick={this.handleItemClick}>
                    <Icon name='file text' color='black' />
                        Details
                    </Menu.Item>
                  </Menu> 
                </Card.Content>
              </Card>
              </Grid.Column>
              <Grid.Column>
              <Card>
              <Label as='a' corner='left' style={{color:'rgb(238,79,76)'}}></Label>
              <Card.Content>
                  <Card.Header style={{paddingLeft:'20px'}}>
                    Reports
                  </Card.Header>
                  <Divider horizontal/>
                    <Image rounded src="/img/Report.jpg" size='small' />
                  <Divider horizontal/>
                </Card.Content>
                <Card.Content >
                  <Menu secondary size='tiny'>
                    <Menu.Item name='report' onClick={this.handleItemClick}>
                    <Icon name='file text' color='black' />
                        Details
                    </Menu.Item>
                  </Menu>
                </Card.Content>
              </Card>
              </Grid.Column>
    
              <Grid.Row centered columns={3}>
              <Grid.Column>
                <Card>
                  <Label as='a' corner='left' style={{color:'rgb(146,41,173)'}}></Label>
                  <Card.Content>
                    <Card.Header style={{paddingLeft:'20px'}}>
                      Query on Reports
                    </Card.Header>
                    <Divider horizontal/>
                      <Image rounded src="/img/Reports.jpg" size='small' />
                    <Divider horizontal/>
                  </Card.Content>
                  <Card.Content >
                    <Menu secondary size='tiny'>
                      <Menu.Item name='query' onClick={this.handleItemClick}>
                      <Icon name='file text' color='black' />
                          Details
                      </Menu.Item>
                    </Menu>
                  </Card.Content>
                </Card>
              </Grid.Column>
              </Grid.Row>
              </Grid>:
              null}
              {activeItem ==='visulize'? 
                <NumberOfCorrectAnswersCategory/>:
              null}
              {activeItem ==='query'? 
                <ReportAnswerCategoryPage/>:
              null}
              {activeItem ==='report'? 
                <SurveyResultsPage/>:
              null}
            </Segment>
      </div>
    );
  }
} 

export { CoverPagesSub as CoverPageSub };

CoverPagesSub.defaultProps = {
    handleItemClick: () => {},
    handleChange: () => {},
    visible: false,
  }

CoverPagesSub.propTypes = {
  handleChange: PropTypes.func.isRequired,
  handleItemClick: PropTypes.func.isRequired,
  visible: PropTypes.bool.isRequired,
}