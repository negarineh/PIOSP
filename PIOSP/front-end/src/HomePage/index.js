export * from './HomePage';
export * from './HomePageSubAdmin';
export * from './CoverPage';
export * from './CoverPageSub';
