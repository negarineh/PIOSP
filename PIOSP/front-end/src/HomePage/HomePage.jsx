import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import '../components/stylesheets/MainPage.css';
import {ReportAnswerCategoryPage} from '../ReportAnswerCategoryPage';
import {ReportAnswerPhotoPage} from '../ReportAnswerCategoryPage';
import {SurveyResultsPage} from '../SurveyResultsPage';
import {SurveyResultsAnswerPage} from '../SurveyResultsPage';
import {AdminList} from '../AdminListPage';
import {AdminTablePage} from '../AdminListPage';
import {ReportAdminPage} from '../ReportAnswerCategoryPage';
import {RegisterPage} from '../RegisterPage';
import FillTable from '../components/Survey/FillTable';
import OptionMenuResults from '../components/Survey/OptionMenuResults';
import {ProfilePage} from '../ProfilePage';

import Collapsible from 'react-collapsible';

import cookie from 'react-cookies';

import { Menu, Segment, Icon, Divider, Grid, Dropdown } from 'semantic-ui-react';
import '../../node_modules/semantic-ui-css/semantic.min.css';
import {CoverPage} from '../HomePage';
import ErrorBoundry from '../ErrorBoundry/ErrorBoundry';


  const tagOptions = [
      {
        text: 'Sign Out',
        value: 'SignOut',
        icon: {name:'sign out', color:'black'},
      },
    ];

// customized dashbord component for main admin
// main admin have access to list of admins and can update, delete or add admins.
class HomePage extends Component {

  constructor(props) {
    super(props);
    this.state = {
      activeItem: 'home',
      visible: false,
      userId:'',
      profileMenu:'',
    };
    this.logoutClick = this.logoutClick.bind(this);
  }

  handleItemClick = (e, { name }) => {
    this.setState({ activeItem: name });
  }


  handleChange = (e, { name, value }) => 
  {
    this.setState({ [name]: value , profileMenu: value});
   
    if (value === 'SignOut')
        this.logoutClick();
  };

  componentDidMount() { 
    this.setState({userId: cookie.load('jwt')});
    cookie.save('jwt', this.state.userId, { path: '/' });
}

logoutClick(){
    cookie.remove('jwt', { path: '/' });
    this.props.history.push('/login');
}

  render() {

      const { activeItem, profileMenu } = this.state;

    return (
      <div >
      
        <Segment style={{paddingTop:"10px", background: 'url("/img/6830481-cloudy-sky.jpg")center center', height:'100vh'}} attached='top'>
          <Menu style={{ height:"20px", background: 'whitesmoke', paddingRight:'20px'}}>
            <Menu.Menu horizontal="true"  position="right" >
              <Menu.Item name='inbox' active={activeItem === 'inbox'} >
                <div className='ui transparent icon input'>
                    <input className='prompt' type='text' placeholder='Search ...' style={{marginTop:"15px", background:'teal'}} />
                    <Icon name='search' style={{marginTop:"2px", paddingRight:'40px'}} color='teal' size='large'/>
                </div>
              </Menu.Item>
              <Dropdown inline icon='user' style={{marginTop:"10px", color:'teal', paddingRight:'20px'}}  direction='left' >
                  <Dropdown.Menu direction='left' >
                    {tagOptions.map(option => <Dropdown.Item key={option.value} {...option} onClick={this.handleChange.bind(this)}/>)}
                    {profileMenu ==='Profile'? <ProfilePage/>: null }
                  </Dropdown.Menu>
              </Dropdown>
            </Menu.Menu>
          </Menu>
         
        
    <Grid  stretched style={{background: 'url("/img/6830481-cloudy-sky.jpg")center center'}}>
      <Grid.Column mobile={4} tablet={4} computer={4} largeScreen={4} widescreen={4}>
        
        <Menu vertical fluid style={{background:'url("/img/Picture4.png")center center', textAlign:'left'}} >
            <Menu.Item name='home' onClick={this.handleItemClick.bind(this)} style={{color:'white'}}>
              <Icon name='home' />
              Home
            </Menu.Item>
            <Divider></Divider> 
            <Menu.Item style={{color:'white'}}>
              <Icon name='users' />
              <Icon name='caret right'/>
              {/* Admins */}
              <Collapsible trigger='Admins'>
              <Menu.Menu>
                <Menu.Item name='List of Admins' onClick={this.handleItemClick.bind(this)} style={{color:'white'}} />
                <Menu.Item name='Edit Admin List' onClick={this.handleItemClick.bind(this)} style={{color:'white'}} />
                <Menu.Item name='Register New Admin' onClick={this.handleItemClick.bind(this)} style={{color:'white'}} />
              </Menu.Menu>
              </Collapsible>
            </Menu.Item>
            <Divider inverted></Divider>
            <Menu.Item name='pollinator' onClick={this.handleItemClick.bind(this)} style={{color:'white'}}>
              <Icon name='database' />
              Pollinator Data
            </Menu.Item>
            <Divider></Divider>
            <Menu.Item name='survey' style={{color:'white'}}>
              <Icon name='server' />
              <Icon name='caret right'/>
              {/* Survey Results */}
              <Collapsible trigger='Survey Results'>
              <Menu.Menu>
                <Menu.Item name='Survey Results(Answer Table)' onClick={this.handleItemClick.bind(this)} style={{color:'white'}} />
                <Menu.Item name='Survey Results(Experience Table)' onClick={this.handleItemClick.bind(this)} style={{color:'white'}} />
              </Menu.Menu>
              </Collapsible>
            </Menu.Item>
            <Divider></Divider>
            <Menu.Item name='report' style={{color:'white'}}>
              <Icon name='external' />
              <Icon name='caret right'/>
              {/* Reports */}
              <Collapsible trigger='Reports'>
              <Menu.Menu>
                <Menu.Item name='Report Answer(Category Base)' onClick={this.handleItemClick.bind(this)} style={{color:'white'}} />
                <Menu.Item name='Report Answer(Photo Name Base)' onClick={this.handleItemClick.bind(this)} style={{color:'white'}} />
                <Menu.Item name='Report Admin List' onClick={this.handleItemClick.bind(this)} style={{color:'white'}} />
              </Menu.Menu>
              </Collapsible>
            </Menu.Item>
            <Divider></Divider>
            <Menu.Item name='visualize' onClick={this.handleItemClick.bind(this)} style={{color:'white'}}>
              <Icon name='pie chart' />
              Visualize Results
            </Menu.Item>
            <Divider></Divider>
            <Menu.Item name='edit' onClick={this.handleItemClick.bind(this)} style={{color:'white'}}>
              <Icon name='edit' />
              Edit Content
            </Menu.Item>
            <Divider></Divider>
            <Menu.Item name='task' onClick={this.handleItemClick.bind(this)} style={{color:'white'}}>
              <Icon name='tasks' />
              Tasks
            </Menu.Item>
        </Menu>
      
      </Grid.Column>
        <Grid.Column stretched mobile={12} tablet={12} computer={12} largeScreen={12} widescreen={12}>
          {activeItem==='home'?
            <ErrorBoundry>
              <CoverPage/>
            </ErrorBoundry>:
            null}
          {activeItem==='List of Admins'?
            <ErrorBoundry>
              <AdminTablePage/>
            </ErrorBoundry>:
            null}
          {activeItem==='Edit Admin List'?
            <ErrorBoundry>  
              <AdminList/>
            </ErrorBoundry>:
            null}
          {activeItem==='Register New Admin'?
            <ErrorBoundry>
              <RegisterPage/>
            </ErrorBoundry>:
            null}
          {activeItem==='pollinator'?
            <ErrorBoundry>  
              <FillTable/>
            </ErrorBoundry>:
            null}
          {activeItem==='Survey Results(Answer Table)'?
            <ErrorBoundry>  
              <SurveyResultsPage/>
            </ErrorBoundry>:
            null}
          {activeItem==='Survey Results(Experience Table)'?
            <ErrorBoundry>  
              <SurveyResultsAnswerPage/>
            </ErrorBoundry>:
            null}
          {activeItem==='Report Answer(Category Base)'?
            <ErrorBoundry>  
              <ReportAnswerCategoryPage/>
            </ErrorBoundry>:
            null}
          {activeItem==='Report Answer(Photo Name Base)'?
              <ReportAnswerPhotoPage/>:
            null}
          {activeItem==='Report Admin List'?
            <ErrorBoundry>  
              <ReportAdminPage/>
            </ErrorBoundry>:
            null}
          {activeItem==='visualize'?
            <ErrorBoundry>
              <OptionMenuResults/>
            </ErrorBoundry>:
            null}
          {activeItem==='edit'?
            <Segment basic>
            </Segment>:
            null}
          {activeItem==='task'?
            <Segment basic>
            </Segment>:
            null}
          </Grid.Column>
        </Grid>
        </Segment>
      </div>
    );
  }
} 

// export default Page3;

function mapStateToProps(state) {
  const { users, authentication } = state;
  const { user } = authentication;
  return {
      user,
      users
  };
}

const connectedHomePage = connect(mapStateToProps)(HomePage);
export { connectedHomePage as HomePage };

HomePage.defaultProps = {
    visible: false,
    handleChange: () => {},
    handleItemClick: () => {},
    logoutClick: () => {},
    history: {}
  }

// prop checks
HomePage.propTypes = {
  visible: PropTypes.bool.isRequired,
  handleChange: PropTypes.func.isRequired,
  handleItemClick: PropTypes.func.isRequired,
  logoutClick: PropTypes.func.isRequired,
  history: PropTypes.object.isRequired
}
