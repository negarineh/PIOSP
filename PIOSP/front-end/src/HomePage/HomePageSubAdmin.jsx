import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import '../components/stylesheets/MainPage.css';
import {ReportAnswerCategoryPage} from '../ReportAnswerCategoryPage';
import {ReportAnswerPhotoPage} from '../ReportAnswerCategoryPage';
import {SurveyResultsPage} from '../SurveyResultsPage';
import {SurveyResultsAnswerPage} from '../SurveyResultsPage';
import {ReportAdminPage} from '../ReportAnswerCategoryPage';
import FillTable from '../components/Survey/FillTable';
import OptionMenuResults from '../components/Survey/OptionMenuResults';

import Collapsible from 'react-collapsible';

import cookie from 'react-cookies';

import { Menu, Segment, Icon, Divider, Grid, Dropdown } from 'semantic-ui-react';
import '../../node_modules/semantic-ui-css/semantic.min.css';
import {CoverPageSub} from '../HomePage';
import ErrorBoundry from '../ErrorBoundry/ErrorBoundry';

  const tagOptions = [
      {
        text: 'Sign Out',
        value: 'SignOut',
        // label: { color: 'black', empty: true, circular: true,  },
        icon: {name:'sign out', color:'black'},
      },
    ];


// customize dashboard for sub admins
// they don't have access to list of admins 
class HomePageSubAdmin extends Component {

  constructor(props) {
    super(props);
    this.state = {
      activeItem: 'home',
      visible: false,
      userId:''
    };
    this.logoutClick = this.logoutClick.bind(this);
  }

  handleItemClick = (e, { name }) => {
    this.setState({ activeItem: name });
  }


  handleChange = (e, { name, value }) => 
  {
    this.setState({ [name]: value });
    if (value === 'Profile')
        alert("Profile");
    if (value === 'SignOut')
        this.logoutClick();
  }

  componentDidMount() { 
    // this.props.dispatch(userActions.getAll());
    // this.state = { userId: cookie.load('jwt') };
    this.setState({userId: cookie.load('jwt')});
    cookie.save('jwt', this.state.userId, { path: '/' });
}

logoutClick(){
    cookie.remove('jwt', { path: '/' });
    this.props.history.push('/login');
}

  render() {

      const { activeItem } = this.state;

    return (
      <div >
      
        <Segment style={{paddingTop:"10px", background: 'url("/img/6830481-cloudy-sky.jpg")center center', height:'100vh'}} attached='top'>
          <Menu style={{ height:"20px", background: 'whitesmoke', paddingRight:'20px'}}>
            <Menu.Menu horizontal="true"  position="right" >
              <Menu.Item name='inbox' active={activeItem === 'inbox'} >
                <div className='ui transparent icon input'>
                    <input className='prompt' type='text' placeholder='Search ...' style={{marginTop:"15px", background:'teal'}} />
                    <Icon name='search' style={{marginTop:"2px", paddingRight:'40px'}} color='teal' size='large'/>
                </div>
              </Menu.Item>
              <Dropdown inline icon='user' style={{marginTop:"10px", color:'teal', paddingRight:'20px'}}  direction='left' >
                  <Dropdown.Menu direction='left' >
                    {tagOptions.map(option => <Dropdown.Item key={option.value} {...option} onClick={this.handleChange.bind(this)}/>)}
                  </Dropdown.Menu>
              </Dropdown>
            </Menu.Menu>
          </Menu>
         
        
    <Grid  stretched style={{background: 'url("/img/6830481-cloudy-sky.jpg")center center'}}>
      <Grid.Column mobile={4} tablet={4} computer={4} largeScreen={4} widescreen={4}>
        
        <Menu vertical fluid style={{background:'url("/img/Picture4.png")center center', textAlign:'left'}} >
            <Menu.Item name='home' onClick={this.handleItemClick.bind(this)} style={{color:'white'}}>
              <Icon name='home' />
              Home
            </Menu.Item>
            <Divider></Divider>
            <Menu.Item name='pollinator' onClick={this.handleItemClick.bind(this)} style={{color:'white'}}>
              <Icon name='database' />
              Pollinator Data
            </Menu.Item>
            <Divider></Divider>
            <Menu.Item name='survey' style={{color:'white'}}>
              <Icon name='server' />
              <Icon name='caret right'/>
              {/* Survey Results */}
              <Collapsible trigger='Survey Results'>
              <Menu.Menu>
                <Menu.Item name='Survey Results(Answer Table)' onClick={this.handleItemClick.bind(this)} style={{color:'white'}} />
                <Menu.Item name='Survey Results(Experience Table)' onClick={this.handleItemClick.bind(this)} style={{color:'white'}} />
              </Menu.Menu>
              </Collapsible>
            </Menu.Item>
            <Divider></Divider>
            <Menu.Item name='report' style={{color:'white'}}>
              <Icon name='external' />
              <Icon name='caret right'/>
              {/* Reports */}
              <Collapsible trigger='Reports'>
              <Menu.Menu>
                <Menu.Item name='Report Answer(Category Base)' onClick={this.handleItemClick.bind(this)} style={{color:'white'}} />
                <Menu.Item name='Report Answer(Photo Name Base)' onClick={this.handleItemClick.bind(this)} style={{color:'white'}} />
                <Menu.Item name='Report Admin List' onClick={this.handleItemClick.bind(this)} style={{color:'white'}} />
              </Menu.Menu>
              </Collapsible>
            </Menu.Item>
            <Divider></Divider>
            <Menu.Item name='visualize' onClick={this.handleItemClick.bind(this)} style={{color:'white'}}>
              <Icon name='pie chart' />
              Visualize Results
            </Menu.Item>
            <Divider></Divider>
            <Menu.Item name='edit' onClick={this.handleItemClick.bind(this)} style={{color:'white'}}>
              <Icon name='edit' />
              Edit Content
            </Menu.Item>
            <Divider></Divider>
            <Menu.Item name='task' onClick={this.handleItemClick.bind(this)} style={{color:'white'}}>
              <Icon name='tasks' />
              Tasks
            </Menu.Item>
        </Menu>
      
      </Grid.Column>
        <Grid.Column stretched mobile={12} tablet={12} computer={12} largeScreen={12} widescreen={12}>
          {activeItem==='home'?
            <ErrorBoundry>
              <CoverPageSub/>
            </ErrorBoundry>:
            null}
          {activeItem==='pollinator'?
            <ErrorBoundry>  
              <FillTable/>
            </ErrorBoundry>:
            null}
          {activeItem==='Survey Results(Answer Table)'?
            <ErrorBoundry>  
              <SurveyResultsPage/>
            </ErrorBoundry>:
            null}
          {activeItem==='Survey Results(Experience Table)'?
            <ErrorBoundry>  
              <SurveyResultsAnswerPage/>
            </ErrorBoundry>:
            null}
          {activeItem==='Report Answer(Category Base)'?
            <ErrorBoundry>  
              <ReportAnswerCategoryPage/>
            </ErrorBoundry>:
            null}
          {activeItem==='Report Answer(Photo Name Base)'?
              <ReportAnswerPhotoPage/>:
            null}
          {activeItem==='Report Admin List'?
            <ErrorBoundry>  
              <ReportAdminPage/>
            </ErrorBoundry>:
            null}
          {activeItem==='visualize'?
            <ErrorBoundry>
              <OptionMenuResults/>
            </ErrorBoundry>:
            null}
          {activeItem==='edit'?
            <Segment basic>
            </Segment>:
            null}
          {activeItem==='task'?
            <Segment basic>
            </Segment>:
            null}
          </Grid.Column>
        {/* </Sidebar.Pushable> */}
        </Grid>
        </Segment>
      </div>
    );
  }
} 

// export default Page3;

function mapStateToProps(state) {
  const { users, authentication } = state;
  const { user } = authentication;
  return {
      user,
      users
  };
}

const connectedHomePage = connect(mapStateToProps)(HomePageSubAdmin);
export { connectedHomePage as HomePageSubAdmin };


HomePageSubAdmin.defaultProps = {
    visible: false,
    handleChange: () => {},
    handleItemClick: () => {},
    logoutClick: () => {},
    history: {}
  }

// prop checks
HomePageSubAdmin.propTypes = {
  visible: PropTypes.bool.isRequired,
  // userId: PropTypes.string.isRequired,
  // activeItem: PropTypes.string.isRequired,
  handleChange: PropTypes.func.isRequired,
  handleItemClick: PropTypes.func.isRequired,
  logoutClick: PropTypes.func.isRequired,
  history: PropTypes.object.isRequired
}
