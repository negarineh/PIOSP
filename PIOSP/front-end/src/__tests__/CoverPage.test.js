import React from 'react';
import {shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme from 'enzyme';
import {CoverPage} from '../HomePage';
import toJson from 'enzyme-to-json';

Enzyme.configure({ adapter: new Adapter()}); 

describe('Cover Page Component', () => {
    describe('render()', () => {
        it('Should render the component', () => {
            const component = shallow(<CoverPage/>);
            const tree = toJson(component);
            console.log(tree);
            // expect(component.contains('Admin Tables')).toBe(true);
            expect(tree).toMatchSnapshot();
        });
    });
});
