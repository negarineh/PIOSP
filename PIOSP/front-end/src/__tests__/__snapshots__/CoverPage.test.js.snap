// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cover Page Component render() Should render the component 1`] = `
<div>
  <Segment
    style={
      Object {
        "background": "whitesmoke",
        "height": "638.5px",
      }
    }
  >
    <Grid
      centered={true}
      columns={3}
      container={true}
    >
      <GridColumn>
        <Card>
          <Label
            as="a"
            corner="left"
            style={
              Object {
                "color": "rgb(254,158,26",
              }
            }
          />
          <CardContent>
            <CardHeader
              style={
                Object {
                  "paddingLeft": "20px",
                }
              }
            >
              Admin Tables
            </CardHeader>
            <Divider
              horizontal={true}
            />
            <Image
              as="img"
              centered={true}
              rounded={true}
              size="small"
              src="/img/table.jpg"
              ui={true}
            />
            <Divider
              horizontal={true}
            />
          </CardContent>
          <CardContent>
            <Menu
              secondary={true}
              size="tiny"
            >
              <MenuItem
                name="admin"
                onClick={[Function]}
              >
                <Icon
                  as="i"
                  color="black"
                  name="file text"
                />
                Details
              </MenuItem>
            </Menu>
          </CardContent>
        </Card>
      </GridColumn>
      <GridColumn>
        <Card>
          <Label
            as="a"
            corner="left"
            style={
              Object {
                "color": "rgb(92,179,96)",
              }
            }
          />
          <CardContent>
            <CardHeader
              style={
                Object {
                  "paddingLeft": "20px",
                }
              }
            >
              Graphs
            </CardHeader>
            <Divider
              horizontal={true}
            />
            <Image
              as="img"
              rounded={true}
              size="small"
              src="/img/images.jpeg"
              ui={true}
            />
            <Divider
              horizontal={true}
            />
          </CardContent>
          <CardContent>
            <Menu
              secondary={true}
              size="tiny"
            >
              <MenuItem
                name="visulize"
                onClick={[Function]}
              >
                <Icon
                  as="i"
                  color="black"
                  name="file text"
                />
                Details
              </MenuItem>
            </Menu>
          </CardContent>
        </Card>
      </GridColumn>
      <GridColumn>
        <Card>
          <Label
            as="a"
            corner="left"
            style={
              Object {
                "color": "rgb(238,79,76)",
              }
            }
          />
          <CardContent>
            <CardHeader
              style={
                Object {
                  "paddingLeft": "20px",
                }
              }
            >
              Reports
            </CardHeader>
            <Divider
              horizontal={true}
            />
            <Image
              as="img"
              rounded={true}
              size="small"
              src="/img/Report.jpg"
              ui={true}
            />
            <Divider
              horizontal={true}
            />
          </CardContent>
          <CardContent>
            <Menu
              secondary={true}
              size="tiny"
            >
              <MenuItem
                name="report"
                onClick={[Function]}
              >
                <Icon
                  as="i"
                  color="black"
                  name="file text"
                />
                Details
              </MenuItem>
            </Menu>
          </CardContent>
        </Card>
      </GridColumn>
      <GridRow
        centered={true}
        columns={3}
      >
        <GridColumn>
          <Card>
            <Label
              as="a"
              corner="left"
              style={
                Object {
                  "color": "rgb(146,41,173)",
                }
              }
            />
            <CardContent>
              <CardHeader
                style={
                  Object {
                    "paddingLeft": "20px",
                  }
                }
              >
                Query on Reports
              </CardHeader>
              <Divider
                horizontal={true}
              />
              <Image
                as="img"
                rounded={true}
                size="small"
                src="/img/Reports.jpg"
                ui={true}
              />
              <Divider
                horizontal={true}
              />
            </CardContent>
            <CardContent>
              <Menu
                secondary={true}
                size="tiny"
              >
                <MenuItem
                  name="query"
                  onClick={[Function]}
                >
                  <Icon
                    as="i"
                    color="black"
                    name="file text"
                  />
                  Details
                </MenuItem>
              </Menu>
            </CardContent>
          </Card>
        </GridColumn>
        <GridColumn>
          <Card>
            <Label
              as="a"
              corner="left"
              style={
                Object {
                  "color": "rgb(24,189,209)",
                }
              }
            />
            <CardContent>
              <CardHeader
                style={
                  Object {
                    "paddingLeft": "20px",
                  }
                }
              >
                Register New Admin
              </CardHeader>
              <Divider
                horizontal={true}
              />
              <Image
                as="img"
                rounded={true}
                size="small"
                src="/img/Register.jpg"
                ui={true}
              />
              <Divider
                horizontal={true}
              />
            </CardContent>
            <CardContent>
              <Menu
                secondary={true}
                size="tiny"
              >
                <MenuItem
                  name="register"
                  onClick={[Function]}
                >
                  <Icon
                    as="i"
                    color="black"
                    name="file text"
                  />
                  Details
                </MenuItem>
              </Menu>
            </CardContent>
          </Card>
        </GridColumn>
      </GridRow>
    </Grid>
  </Segment>
</div>
`;
